### PR TITLE
Agnostification of gadget-common and introduction of GadgetEnvironments

### PR DESCRIPTION
**Pull request by tbraun96**

### CODE REVIEW

File: common/src/client.rs [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
+use sp_core::Pair;
 use std::sync::Arc;
-use tangle_subxt::subxt::{self, tx::TxPayload, OnlineClient};
+use tangle_subxt::subxt::{self};
 
 pub struct JobsClient<Env: GadgetEnvironment> {
     pub client: Arc<Env::Client>,
     logger: DebugLogger,
-    pallet_tx: Arc<dyn PalletSubmitter>,
+    pub(crate) pallet_tx: Env::TransactionManager,
```
Comment (drewstone) [inline common/src/client.rs:14]: Pallet tx isn't the right term, maybe tx_manager?

**Review by drewstone — CHANGES_REQUESTED** [review]

**Review by drewstone — CHANGES_REQUESTED** [review]
We haven't moved to v2 yet, this is taking a bit longer than it likely needs to. @tbraun96 i really recommend shifting to jobs/v2 api soon. I'm not seeing it in this PR.

**Comment by tbraun96** [comment]
> We haven't moved to v2 yet, this is taking a bit longer than it likely needs to. @tbraun96 i really recommend shifting to jobs/v2 api soon. I'm not seeing it in this PR.

Completely re-organizing and agnostifying the entire codebase after we decided on a model change is not easy. These PRs are indeed v2-related, since it will allow us to define the v2 calls seamlessly once complete. Without these PRs, we have no v2 potential

File: Cargo.toml [hunk 2]
```diff
@@ -40,7 +44,7 @@ tangle-subxt = { git = "https://github.com/webb-tools/tangle.git", default-featu
 #pallet-dkg = { path = "../tangle/pallets/dkg", default-features = false }
```
Comment (drewstone) [inline Cargo.toml:44]: We can delete this

File: Cargo.toml [hunk 3]
```diff
@@ -40,7 +44,7 @@ tangle-subxt = { git = "https://github.com/webb-tools/tangle.git", default-featu
...
 #pallet-zksaas = { path = "../tangle/pallets/zksaas", default-features = false }
```
Comment (drewstone) [inline Cargo.toml:45]: We can delete this

File: Cargo.toml [hunk 4]
```diff
@@ -40,7 +44,7 @@ tangle-subxt = { git = "https://github.com/webb-tools/tangle.git", default-featu
...
 #tangle-primitives = { path = "../tangle/primitives", default-features = false }
-#tangle-subxt = { path = "../tangle/tangle-subxt", default-features = false }
+#tangle-transactions = { path = "../tangle/tangle-transactions", default-features = false }
```
Comment (drewstone) [inline Cargo.toml:47]: What is this? Do these other things still need to be commented out?

File: common/src/lib.rs [hunk 5]
```diff
@@ -221,11 +224,16 @@ pub async fn create_work_manager<Env: GadgetEnvironment, P: GadgetProtocol<Env>>
...
-    })
+    client: &<Env as GadgetEnvironment>::Client,
+) -> Result<<Env as GadgetEnvironment>::Event, Error>
+where
+    <Env as GadgetEnvironment>::Client: Client<<Env as GadgetEnvironment>::Event>,
+{
+    Client::<Env::Event>::latest_event(client)
+        .await
+        .ok_or_else(|| Error::InitError {
+            err: "No finality notification received".to_string(),
```
Comment (drewstone) [inline common/src/lib.rs:235]:
```suggestion
            err: "No event received".to_string(),
```

File: environments/tangle/src/client.rs [hunk 6]
```diff
@@ -0,0 +1,201 @@
+/*use gadget_common::client::{ClientWithApi, JobsClient, JobTypeExt, PhaseResultExt};
```
Comment (drewstone) [inline environments/tangle/src/client.rs:1]: Why is this file commented out? Can it be removed?

**Review by drewstone — CHANGES_REQUESTED** [review]

**Pull request merged**